Unbox.swift

```diff
@@ -638,42 +633,94 @@ extension UnboxValueResolver where T: CollectionType, T: DictionaryLiteralConver
             
-            return Unboxer(dictionary: dictionary, context: context)
+            return Unboxer(dictionary: dictionary, context: context, allowInvalidElements: allowInvalidElements)
         } catch {
             throw UnboxError.InvalidData
         }
     }
     
-    static func unboxersFromData(data: NSData, context: Any?) throws -> [Unboxer] {
+    static func unboxersFromData(data: NSData, context: Any?, allowInvalidElements: Bool) throws -> [Unboxer] {
         do {
             guard let array = try NSJSONSerialization.JSONObjectWithData(data, options: [.AllowFragments]) as? [UnboxableDictionary] else {
                 throw UnboxError.InvalidData
             }
             
             return array.map({
-                return Unboxer(dictionary: $0, context: context)
+                return Unboxer(dictionary: $0, context: context, allowInvalidElements: allowInvalidElements)
             })
         } catch {
             throw UnboxError.InvalidData
         }
     }
+
+    static func performUnboxingWithDictionary<T: Unboxable>(dictionary: UnboxableDictionary, context: Any? = nil, allowInvalidElements: Bool = false) throws -> T {
+        return try Unboxer(dictionary: dictionary, context: context, allowInvalidElements: allowInvalidElements).performUnboxing()
+    }
+
+    static func performUnboxingWithDictionary<T: Unboxable>(dictionaries: [UnboxableDictionary], context: Any? = nil, allowInvalidElements: Bool = false) throws -> [T] {
+        if allowInvalidElements {
+            return dictionaries.flatMap {
+                try? performUnboxingWithDictionary($0, context: context, allowInvalidElements: allowInvalidElements)
+            }
+        }
+        
+        return try dictionaries.map {
+            try performUnboxingWithDictionary($0, context: context)
+        }
+    }
+
+    static func performUnboxingWithData<T: Unboxable>(data: NSData, context: Any? = nil, allowInvalidElements: Bool = false) throws -> T {
+        return try Unboxer.unboxerFromData(data, context: context, allowInvalidElements: allowInvalidElements).performUnboxing()
+    }
+
+    static func performUnboxingWithData<T: Unboxable>(data: NSData, context: Any? = nil, allowInvalidElements: Bool = false) throws -> [T] {
+        return try Unboxer.unboxersFromData(data, context: context, allowInvalidElements: allowInvalidElements).map {
+            return try $0.performUnboxing()
+        }
+    }
+
+    static func performUnboxingWithDictionaryAndContext<T: UnboxableWithContext>(dictionary: UnboxableDictionary, context: T.ContextType, allowInvalidElements: Bool = false) throws -> T {
+        return try Unboxer(dictionary: dictionary, context: context, allowInvalidElements: allowInvalidElements).performUnboxingWithContext(context)
+    }
+
+    static func performUnboxingWithDictionaryAndContext<T: UnboxableWithContext>(dictionaries: [UnboxableDictionary], context: T.ContextType, allowInvalidElements: Bool = false) throws -> [T] {
```

Comment:
The method name here should reflect that it takes `Dictionaries` instead of just `Dictionary`